specification/iothub/resource-manager/Microsoft.Devices/stable/2018-04-01/iothub.json

```diff
@@ -0,0 +1,2664 @@
+          "type": "string",
+          "readOnly": true
+        },
+        "hostName": {
+          "description": "The name of the host.",
+          "type": "string",
+          "readOnly": true
+        },
+        "eventHubEndpoints": {
+          "description": "The Event Hub-compatible endpoint properties. The possible keys to this dictionary are events and operationsMonitoringEvents. Both of these keys have to be present in the dictionary while making create or update calls for the IoT hub.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/EventHubProperties"
+          }
+        },
+        "routing": {
+          "$ref": "#/definitions/RoutingProperties"
+        },
+        "storageEndpoints": {
+          "description": "The list of Azure Storage endpoints where you can upload files. Currently you can configure only one Azure Storage account and that MUST have its key as $default. Specifying more than one storage account causes an error to be thrown. Not specifying a value for this property when the enableFileUploadNotifications property is set to True, causes an error to be thrown.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/StorageEndpointProperties"
+          }
+        },
+        "messagingEndpoints": {
+          "description": "The messaging endpoint properties for the file upload notification queue.",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/MessagingEndpointProperties"
+          }
+        },
+        "enableFileUploadNotifications": {
+          "description": "If True, file upload notifications are enabled.",
+          "type": "boolean"
+        },
+        "cloudToDevice": {
+          "$ref": "#/definitions/CloudToDeviceProperties"
+        },
+        "comments": {
+          "description": "IoT hub comments.",
+          "type": "string"
+        },
+        "operationsMonitoringProperties": {
+          "$ref": "#/definitions/OperationsMonitoringProperties"
+        },
+        "features": {
+          "description": "The capabilities and features enabled for the IoT hub.",
+          "enum": [ "None", "DeviceManagement" ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "Capabilities",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "IotHubSkuInfo": {
+      "description": "Information about the SKU of the IoT hub.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name of the SKU.",
+          "enum": [ "F1", "S1", "S2", "S3", "B1", "B2", "B3" ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "IotHubSku",
+            "modelAsString": true
+          }
+        },
+        "tier": {
+          "description": "The billing tier for the IoT hub.",
+          "enum": [ "Free", "Standard", "Basic" ],
```

Comment:
added "Basic" to enum with `modelAsString: false` => breaking change

Reply:
@olydis - Yes this is a breaking change since modelAsString is set to false. This is my understanding of code gen. Shouldnt this be set to true for enums and linter catch it in case its not. I mean should we even allow it to be set to false?